controllers/validation/validation.go

```diff
@@ -105,6 +105,10 @@ func validateArbiterSpec(mdb mdbv1.MongoDBCommunity) error {
 func validateAuthModeSpec(mdb mdbv1.MongoDBCommunity) error {
 	allModes := mdb.Spec.Security.Authentication.Modes
 
+	if len(allModes) == 0 {
```

Comment:
We already have a default mode set here: https://github.com/mongodb/mongodb-kubernetes-operator/blob/master/api/v1/mongodbcommunity_types.go#L51

I think it would be better to make sure the default is used whenever the mode is not specified. @mircea-cosbuc WDYT?

Reply:
Using the default makes sense to me. We would have to add some tests for the consistency of behaviour.

Reply:
@hoyhbx Could I ask to introduce the changes than?

Reply:
Is the desired behavior to set the Modes to the default mode or rejecting the value?
If we want to change the empty Modes to the default mode, I am not sure if this validation function is the correct place to fix this, since all the validation functions seem only validates but not mutates the CR.

BTW, we realized this validation could be done on the CRD side, by adding the 
`// +kubebuilder:validation:MinItems:=1` to https://github.com/mongodb/mongodb-kubernetes-operator/blob/d1e7ba4bf458c6ae4d20a50dd9fd068b446dd6d4/api/v1/mongodbcommunity_types.go#L517